bugbug/tools/code_review.py

```diff
@@ -978,12 +1108,20 @@ def vector_points():
     def find_similar_hunk_comments(self, hunk: Hunk):
         return self.vector_db.search(self.embeddings.embed_query(str(hunk)))
 
-    def find_similar_patch_comments(self, patch: Patch):
+    def find_similar_patch_comments(self, patch: Patch, limit: int):
+        assert limit > 0, "Limit must be greater than 0"
+
         patch_set = PatchSet.from_string(patch.raw_diff)
 
+        yielded = 0
         for patched_file in patch_set:
             if not patched_file.is_modified_file:
                 continue
 
             for hunk in patched_file:
-                yield from self.find_similar_hunk_comments(hunk)
+                for result in self.find_similar_hunk_comments(hunk):
+                    yield result
+
+                    yielded += 1
+                    if yielded >= limit:
+                        return
```

Comment:
In this case, search for one examples per hunk in order of the patch and limit to 10 examples (according to the call), this will not scale with bigger reviews, where later hunks/files will be ignored. I would generate (at least) one for each hunk then report the top [:limit] ordered by decreasing similarity from the cosine distance that is used. By taking the top, it should also show examples that are similar to code changes that actually need reviews, as they have similar examples in the db

Reply:
I agree with you. I have a todo regarding this: https://github.com/mozilla/bugbug/blob/3aea837c99613eac22e3c859499f667278409515/bugbug/tools/code_review.py#L1030-L1031

I will handle it in a separate pull request.